src/database.c

```diff
@@ -31,185 +31,179 @@ sqlite3 *DB;
 }
 
-int log_file_op(char *operation, char *path, size_t size, size_t usage, char* rstatus, int errorCode) {
-        /* SQL commands to compile */
-        const char *sql = "CREATE TABLE if not exists Logs(\
-                               Time TEXT,\ 
-                               UID INT,\ 
-                               Operation TEXT,\
-                               Size INT,\
-                               Status TEXT,\
-                               errorCode INT,\
-                               Path TEXT);\ 
-                           INSERT INTO Logs\
-                               VALUES(%s, %d, %s, %d, %s,%d, %s);";  
-        char *err = NULL;
-        char *sqlbuf = NULL;
-        char *timebuf = NULL;
-        int uid;
-        int updateSize;
-        time_t now;
-        int rc,rc1;
-
-        /* allocate heap data and check for succcess */
-        sqlbuf = malloc(BUF_MAX);
-        if (!sqlbuf) {
-                fprintf(stderr,"ERROR: couldn't allocate SQL buffer\n");
-                return -ENOMEM;
-        }
-
-        timebuf = malloc(TIME_MAX);
-        if (!timebuf) {
-                fprintf(stderr,"ERROR: couldn't allocate Time buffer\n");
-                return -ENOMEM;
-        }
-        /* get user id */
-        uid = getuid();  
-
-        /* get current time and convert it to readable */
-        time(&now);
-        strftime(timebuf, TIME_MAX, "%c",localtime(&now)); 
-
-        /* read information into buffer s */
-        sprintf(sqlbuf, sql, addquote(timebuf),
-                        uid, addquote(operation), 
-                        size, addquote(rstatus),errorCode,addquote(path));
-
-        /* execute the SQL statement & check for success*/
-        rc = sqlite3_exec(DB, sqlbuf, NULL, NULL, &err);
-        if (rc != SQLITE_OK) {
-                fprintf(stderr, "SQL error: %s\n", err);
-                sqlite3_free(err);
-        }
-
-        /* update usage */
-        updateSize = (int) usage;
-        
-        /* check for growth in usage */
-        if (strcmp(operation,"Write") == 0 || strcmp(operation,"Mkdir") == 0) {
-                updateQuotas(timebuf,uid,updateSize);
-        }
-
-        /*check for shrinking usage */
-        if(strcmp(operation,"Unlink") == 0 || strcmp(operation,"Rmdir") == 0) {
-                updateSize *= -1;
-                updateQuotas(timebuf,uid,updateSize);
-	
-        }
-        /* free all heap data */
-        free(sqlbuf);
-        free(timebuf);
-        return rc;
+int updateQuotas(char* time,int uid, int size )
+{
+    char *sql = NULL;
+    char *sqlbuf = NULL;
+    char *err = NULL;
+    int rc;
+    /* allocate heap data and check for succcess */
+    sqlbuf = malloc(BUF_MAX);
+    if (!sqlbuf) {
+        fprintf(stderr,"ERROR: couldn't allocate SQL buffer\n");
+        return -ENOMEM;
+    }
+    if (size >= 0) {
+        sql = "INSERT INTO Quotas(Time,UID, Usage, Quota) VALUES\
+		(%s,%d,%d,%d-%d) ON CONFLICT(UID) DO UPDATE SET \
```

Comment:
Just curious, do we need to update quota value? I can not recollect the database table and the logic behind updating the quota value here. It makes sense to update the usage value. Just to back up my previous comment, it helps to understand the logic better if you include the comments around logic like this.

Reply:
Here we are actually updating the usage, and the quotas is always equal (initial quota - usage), like initial quota is 10000, and usage is 0. When the usage become 1000, the quota should automatically become 9000

Reply:
Got it, will do